### PR TITLE
Connect device with a certificate

### LINKED ISSUE #15: Client can authentication with certificates

### PR DESCRIPTION
**Pull request by Ioana37**
Closes #15

### CODE REVIEW

**Review by vjrantal — COMMENTED** [review]
PR looks good, but if possible, please check if the new scripts works in our devcontainer.

File: Testing/MqttTests.cs [hunk 1]
```diff
@@ -15,6 +16,8 @@ public class MqttTests
     {
         private string iotHubConnectionString;
         private string iotHubDeviceConnectionString;
+        private string iotHubDeviceCertifConnectionString;
```
Comment (syedhassaanahmed) [inline Testing/MqttTests.cs]: Nitpicking here :) but can we use either "Certificate" or "Cert" in the variable names?

File: Device/Device.cs [hunk 2]
```diff
@@ -66,7 +69,43 @@ public async Task DisconnectDevice()
...
+                })
+                .Build();
+
+            await mqttClient.ConnectAsync(options, CancellationToken.None);
+        }
+
+        public async Task ConnectDeviceUsingSelfSignedCertificate()
+        {
+            var certificates = new List<X509Certificate>() { new X509Certificate2("SelfSigned-Certificate.pfx", "1234") };
+            var options = new MqttClientOptionsBuilder()
```
Comment (EliiseS) [inline Device/Device.cs]: The options above and here are very similar, perhaps, we should make a method that accepts certificate as a parameter?

File: docs/CACertificates/CACertificateForDevice.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+## Introduction
+
+The tools in this file can be used to setup CA Certificates (along with proof of possession).
+
+This file contains a Bash script to help create **test** certificates for Azure IoT Hub's CA Certificate / proof-of-possession.
+A more detailed document showing UI screen shots for CA Certificates and proof of possession flow is available from [the official documentation].
+
+## USE
+
+## Step 1 - Initial Setup
```
Comment (EliiseS) [inline docs/CACertificates/CACertificateForDevice.md]: Should the steps be `###`?

```suggestion
### Step 1 - Initial Setup
```

File: docs/CACertificates/CACertificateForDevice.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# Managing test CA certificates for connecting a device to the IoT Hub
```
Comment (EliiseS) [inline docs/CACertificates/CACertificateForDevice.md]: Should you reference this file from the contributing.md?

**Comment by Ioana37** [comment]
> PR looks good, but if possible, please check if the new scripts works in our devcontainer.

I checked and they work.

**Review by syedhassaanahmed — APPROVED** [review]

**Review by EliiseS — APPROVED** [review]

**Pull request merged**